src/main/java/com/direwolf20/buildinggadgets/common/building/modes/AbstractMode.java

```diff
@@ -0,0 +1,43 @@
+package com.direwolf20.buildinggadgets.common.building.modes;
+
+import com.direwolf20.buildinggadgets.common.building.Context;
+import com.direwolf20.buildinggadgets.common.building.IBuildingMode;
+import com.direwolf20.buildinggadgets.common.building.IValidatorFactory;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+import java.util.function.BiPredicate;
+
+/**
+ * Base class for Building Gadget's native mode implementations.
+ */
+abstract class AbstractMode implements IBuildingMode {
```

Comment:
I think this should be public, so that people have a starting point and don't need to implement everything themselves.
And this of course needs Documentation

Reply:
`...to allow reuse validator implementation` is the only thing that I can think of for extra documentation.